src/main/java/io/coala/jetbrains/settings/ProjectSettings.java

```diff
@@ -0,0 +1,64 @@
+package io.coala.jetbrains.settings;
+
+import com.intellij.openapi.components.ProjectComponent;
+import com.intellij.openapi.project.Project;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProjectSettings implements ProjectComponent {
+    private final Project project;
+    private String cwd;
+    private String executable;
+    private List<String> sections;
+
+    public ProjectSettings(Project project) {
+        this.project = project;
+        this.cwd = project.getBasePath();
+        /* TODO: handle other OSes
+         *  in the future this value will be set from the settings panel
+         */
+        this.executable = null;
+        this.sections = new ArrayList<>();
+    }
+
+    public Project getProject() {
+        return project;
+    }
+
+    public String getCwd() {
+        return cwd;
+    }
+
+    public String getExecutable() {
+        return executable;
+    }
+
+    public List<String> getSections() {
+        return sections;
+    }
+
+    public void setExecutable(String executable) {
+        this.executable = executable;
+    }
+
+    public void addSections(List<String> sections) {
```

Comment:
What does a `section` represent here?

Reply:
coala `section_tags` value

Reply:
Maybe call it with a more appropriate name, something like `addSectionToFilter`.